pkg/controller/uniteddeployment/uniteddeployment_controller.go

```diff
@@ -257,3 +259,115 @@ func (r *ReconcileUnitedDeployment) classifySubsetBySubsetName(ud *appsv1alpha1.
 	}
 	return mapping
 }
+
+func (r *ReconcileUnitedDeployment) updateStatus(instance *appsv1alpha1.UnitedDeployment, oldStatus *appsv1alpha1.UnitedDeploymentStatus, nameToSubset *map[string]*Subset, nextReplicas, nextPartition *map[string]int32, currentRevision, updatedRevision *appsv1.ControllerRevision, collisionCount int32) (reconcile.Result, error) {
+	newStatus := r.calculateStatus(instance, nameToSubset, nextReplicas, nextPartition, currentRevision, updatedRevision, collisionCount)
+	_, err := r.updateUnitedDeployment(instance, oldStatus, newStatus)
+	return reconcile.Result{}, err
+}
+
+func (r *ReconcileUnitedDeployment) calculateStatus(ud *appsv1alpha1.UnitedDeployment, nameToSubset *map[string]*Subset, nextReplicas, nextPartition *map[string]int32, currentRevision, updatedRevision *appsv1.ControllerRevision, collisionCount int32) *appsv1alpha1.UnitedDeploymentStatus {
+	newStatus := ud.Status
+
+	expectedRevision := currentRevision.Name
+	if updatedRevision != nil {
+		expectedRevision = updatedRevision.Name
+	}
+
+	var replicas int32
```

Comment:
Do we need these four temporal variables?

Reply:
done